### PR TITLE
Adds documentation

### PR DESCRIPTION
**Pull request by tylermeekel**
This pull request adds missing documentation.

### Progress
- [x] template package
- [x] multiInput and textInput packages
- [x] steps package
- [x] program package
- [x] utils package
- [x] rest of cmd package

### CODE REVIEW

**Comment by Melkeydev** [comment]
This looks really good - excited to see it finished

**Comment by tylermeekel** [comment]
Should be good to go. There were some things that I wasn't 100% sure of that might need a revision.

File: cmd/program/program.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// Package program contains utility that help
```
Comment (Melkeydev) [inline cmd/program/program.go:1]: Nit: Its not necessarily utility, program is kind of the back bone of the entire logic for the CLI

**Review by Melkeydev — CHANGES_REQUESTED** [review]
Added a nit on one of the comments

File: cmd/program/program.go (hunk 1 above)
Comment (tylermeekel) [reply cmd/program/program.go:1]: Yeah that was definitely a late night moment. I'll rewrite it quick.

**Comment by Melkeydev** [comment]
You have a merge conflict that needs to be resolved before merging!

**Pull request merged**